### ISSUE TITLE
Some key events (e.g. arrows) ignored in shell inside a container, starting with batect 0.80.0

### ISSUE THREAD

**Issue by lbv** [batect/batect#1392]
### Describe the bug

Hi! I've used Batect from time to time to tinker with development environments in Windows, and after a while without meddling with it (because where I've used it, it just works™), I installed a recent version of batect to prepare a new workspace, but noticed something strange.

If I prepare a task that simply starts a shell inside a container, I can certainly start the shell but then I can't use certain keys like the arrows, Home or End, which makes it pretty uncomfortable to test things inside the container. For example, I can't easily move through the commands inside bash's history.

After trying many things (including updating Docker for windows, git for windows, windows itself to windows 11, java, and trying different terminal emulators and like windows terminal, cmder, alacritty, and shells like CMD, git-bash, powershell), nothing has fixed the issue, but I noticed that the problem only shows itself in versions >= 0.80.0 of batect.

### Steps to reproduce

Create a new workspace, with a simple `batect.yml` that has a task to simply start a shell, like this:

```
containers:
  ubu:
    image: ubuntu:22.04

tasks:
  sh:
    run:
      container: ubu
      command: bash
```

From windows terminal, start the task:

```
./batect sh
```

Run commands inside the shell, and then try to move through bash's history (up arrow, down arrow) or move the cursor on the command you're typing (left arrow, right arrow, home)

### Expected behaviour

Up arrow, down arrow: move through history
Left arrow, right arrow: move the cursor through the commands

### Actual behaviour

The arrows don't do anything

### Version information

```plain text
./batect --version
Batect version:    0.82.0
Built:             2022-09-26 07:28:01 +0000
Built from commit: 7e42c7521096d3fd65fd06be4725a0fb1f7488f2 (commit date: 2022-09-26 16:45:03 +1000)
JVM version:       Azul Systems, Inc. OpenJDK 64-Bit Server VM 11.0.16.1
OS version:        Windows 11 10.0 amd64
Docker version:    20.10.21 (API version: 1.41, minimum supported API version: 1.12, commit: 3056208, operating system: 'linux', architecture: 'amd64', experimental: false)
Git version:       2.38.1.windows.1

For documentation and further information on Batect, visit https://github.com/batect/batect.
```
```


### Any other information

I've checked that this problem occurs in versions of Batect >= 0.80.0

Batect 0.79.1 works as expected.

**Comment by charleskorn** [comment]
Thanks for the bug report @lbv. 

If you run `docker run -it ubuntu:22.04 bash` from a terminal, do you see the same behaviour?

**Comment by lbv** [comment]
> If you run `docker run -it ubuntu:22.04 bash` from a terminal, do you see the same behaviour?

No. Meaning, when I run `docker run ...` directly, it works as expected regarding the arrow keys and others like Home, End, etc.

**Comment by charleskorn** [comment]
That definitely sounds like a bug in Batect then. 

I'm currently travelling, but I will take a look at this when I return home in a few weeks.

**Comment by charleskorn** [comment]
This should be fixed in [0.83.0](https://github.com/batect/batect/releases/tag/0.83.0).
